c/parallel/thread_pool.cc

```diff
@@ -132,28 +153,33 @@ thread_team* thread_pool::get_team_unchecked() noexcept {
 }
 
 
-void thread_pool::enable_monitor(bool a) const noexcept {
-  controller.enable_monitor(a);
+void thread_pool::enable_monitor(bool a) noexcept {
+  init_monitor_thread();
+  monitor->set_active(a);
 }
 
 
-bool thread_pool::is_monitor_enabled() const noexcept {
-  return controller.is_monitor_enabled();
+bool thread_pool::is_monitor_enabled() noexcept {
+  init_monitor_thread();
+  return monitor->get_active();
```

Comment:
don't need to init monitor here: if there is no monitor then it is definitely "not enabled"

Reply:
Yeah, forgot to add just a simple check here.